src/main/helpers.js

```diff
@@ -310,9 +312,12 @@ export function insertContentType(metaData, filePath) {
   return metaData
 }
 
-// Function prepends metadata with the appropriate prefix if it is not already on
-export function prependXAMZMeta(metaData) {
+// Function normalizes the metadata
+export function cleanMetadata(metaData) {  
+  // Prepends metadata with the appropriate prefix if it is not already on
   var newMetadata = Object.assign({}, metaData)
+  // URL escape the values  
+  _.map(metaData, (v, k) => (!isSupportedHeader(key) && !isStorageclassHeader(key)) ? metaData[k] = querystring.escape(v): metaData[k] = v)  
```

Comment:
What does `key` here refer to? Should this be `k` instead?

I got a few test failures at the `isSupportedHeader` function with the error `TypeError: Cannot read property 'toLowerCase' of undefined`.  Here's the stack trace associated with our library:
```
at toLowerCase (dist/main/helpers.js:344:41)
at isSupportedHeader (dist/main/helpers.js:320:31)
```